packages/mds-agency/request-handlers.ts

```diff
@@ -278,8 +278,12 @@ export const submitVehicleEvent = async (req: AgencyApiRequest, res: AgencyApiRe
     try {
       await cache.readDevice(event.device_id)
     } catch (err) {
-      await Promise.all([cache.writeDevice(device), stream.writeDevice(device)])
-      logger.info('Re-adding previously deregistered device to cache', err)
+      try {
+        await Promise.all([cache.writeDevice(device), stream.writeDevice(device)])
```

Comment:
Not crazy about nested `try/catch`, but since the cache throws on a miss it seems unavoidable.

Could we add some logic to check for a cache miss error specifically?

Reply:
Having the cache (or db) throw on "misses" often results in the need for these sorts of calisthenics. I can think of use cases where a miss is the _desired_ outcome in which case throwing an error is the opposite of what you really want. It's for this reason that I generally prefer returning `null` or `undefined` and letting the caller decide the appropriate course of action.